From e1e97f44c7638b8fdd6362619c088494f9896d63 Mon Sep 17 00:00:00 2001
From: Andreas Oberritter <obi@opendreambox.org>
Date: Fri, 8 Mar 2013 22:06:55 +0100
Subject: [PATCH] sysdeputil: allow login with empty passwords in /etc/passwd
 or /etc/shadow

Signed-off-by: Andreas Oberritter <obi@opendreambox.org>
---
 sysdeputil.c |    8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/sysdeputil.c b/sysdeputil.c
index 06f01f4..c003af1 100644
--- a/sysdeputil.c
+++ b/sysdeputil.c
@@ -284,6 +284,10 @@ vsf_sysdep_check_auth(struct mystr* p_user_str,
       {
         return 0;
       }
+      if (!p_spwd->sp_pwdp[0])
+      {
+        return 1;
+      }
       p_crypted = crypt(str_getbuf(p_pass_str), p_spwd->sp_pwdp);
       if (!vsf_sysutil_strcmp(p_crypted, p_spwd->sp_pwdp))
       {
@@ -292,6 +296,10 @@ vsf_sysdep_check_auth(struct mystr* p_user_str,
     }
   }
   #endif /* VSF_SYSDEP_HAVE_SHADOW */
+  if (!p_pwd->pw_passwd[0])
+  {
+    return 1;
+  }
   p_crypted = crypt(str_getbuf(p_pass_str), p_pwd->pw_passwd);
   if (!vsf_sysutil_strcmp(p_crypted, p_pwd->pw_passwd))
   {
-- 
1.7.10.4

