From 910377538f51b84c0f9e7ed92582d6034b0e88dc Mon Sep 17 00:00:00 2001
From: Andreas Monzner <andreas.monzner@dream-property.net>
Date: Tue, 5 Mar 2013 22:23:21 +0100
Subject: [PATCH 11/17] Keep auto-installed recommends when moved from depends
 to recommends

---
 libopkg/opkg_install.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/libopkg/opkg_install.c b/libopkg/opkg_install.c
index be36a64..8cd5cbf 100644
--- a/libopkg/opkg_install.c
+++ b/libopkg/opkg_install.c
@@ -366,11 +366,12 @@ pkg_remove_orphan_dependent(pkg_t *pkg, pkg_t *old_pkg)
 
 			for (k=0; k<count1; k++) {
 				cd1 = &pkg->depends[k];
-				if (cd1->type != DEPEND)
+				if (cd1->type != DEPEND && cd1->type != RECOMMEND)
 					continue;
 				for (l=0; l<cd1->possibility_count; l++) {
-					if (cd0->possibilities[j]
-					 == cd1->possibilities[l]) {
+					if ( (cd0->type == cd1->type && cd0->possibilities[j] == cd1->possibilities[l]) ||
+					     (cd0->type != cd1->type && !strcmp(cd0->possibilities[j]->pkg->name, cd1->possibilities[l]->pkg->name) )
+					   ) {
 						found = 1;
 						break;
 					}
-- 
1.8.1.2

