From 8b5564f0ff75bd65b480ac37809868df7892cda3 Mon Sep 17 00:00:00 2001
From: Andreas Oberritter <obi@opendreambox.org>
Date: Sun, 17 Mar 2013 22:51:13 +0100
Subject: [PATCH 12/19] is_pkg_a_replaces(): simplify code
To: opkg-devel@googlegroups.com
Cc: paul@paulbarker.me.uk

Signed-off-by: Andreas Oberritter <obi@opendreambox.org>
---
 libopkg/pkg_depends.c | 10 +++-------
 1 file changed, 3 insertions(+), 7 deletions(-)

diff --git a/libopkg/pkg_depends.c b/libopkg/pkg_depends.c
index 2422fb5..6b5a1a5 100644
--- a/libopkg/pkg_depends.c
+++ b/libopkg/pkg_depends.c
@@ -345,22 +345,18 @@ pkg_hash_fetch_satisfied_dependencies(pkg_t * pkg)
   really conflicts
   returns 0 if conflicts <> replaces or 1 if conflicts == replaces
 */
-static int
-is_pkg_a_replaces(pkg_t *pkg_scout,pkg_t *pkg)
+static int is_pkg_a_replaces(const pkg_t *pkg_scout, const pkg_t *pkg)
 {
     int i ;
-    int replaces_count = pkg->replaces_count;
 
-    if (pkg->replaces_count==0)    // No replaces, it's surely a conflict
-        return 0;
-
-    for (i = 0; i < replaces_count; i++) {
+    for (i = 0; i < pkg->replaces_count; i++) {
         if (strcmp(pkg_scout->name,pkg->replaces[i]->name)==0) {      // Found
             opkg_msg(DEBUG2, "Seems I've found a replace %s %s\n",
 			pkg_scout->name, pkg->replaces[i]->name);
             return 1;
         }
     }
+
     return 0;
 
 }
-- 
1.9.1

