From 6d6c96e66f1b8abdb6c4fc45856a8b4eb0396f38 Mon Sep 17 00:00:00 2001
From: ghost <andreas.monzner@dream-property.net>
Date: Wed, 7 Dec 2016 11:51:37 +0100
Subject: [PATCH] add libusbtuner.so to LD_PRELOAD when available

---
 /usr/bin/enigma2-environment | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff -Naur a/usr/bin/enigma2-environment b/usr/bin/enigma2-environment
--- a/usr/bin/enigma2-environment	2016-11-23 16:45:38.000000000 +0100
+++ b/usr/bin/enigma2-environment	2016-12-07 12:46:29.731290019 +0100
@@ -3,7 +3,7 @@
 mkdir -p /run/enigma2
 
 LD_PRELOAD=""
-for lib in libopen.so libpagecache.so libpassthrough.so; do
+for lib in libopen.so libpagecache.so libusbtuner.so libpassthrough.so; do
 	if [ -f /usr/lib/${lib} ]; then
 		LD_PRELOAD="${LD_PRELOAD} /usr/lib/${lib}"
 	fi
-- 
2.1.4
