From 6bb37faac663118a796b88466e9fab20d0140c5a Mon Sep 17 00:00:00 2001
From: Andreas Monzner <andreas.monzner@dream-property.net>
Date: Tue, 15 Jan 2013 16:09:55 +0100
Subject: [PATCH 2/2] dreambox_nand.c: dm8000 speedup flash read a bit..

---
 drivers/mtd/nand/dreambox_nand.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/mtd/nand/dreambox_nand.c b/drivers/mtd/nand/dreambox_nand.c
index 3aff2a9..08f694b 100644
--- a/drivers/mtd/nand/dreambox_nand.c
+++ b/drivers/mtd/nand/dreambox_nand.c
@@ -264,6 +264,8 @@ static int __init dreambox_nand_init(void)
 		return -ENOMEM;
 	}
 
+	writel(0x00000481, (void __iomem*)0xB000080c);
+
 	/* Get pointer to private data */
 	this = (struct nand_chip *)&dreambox_mtd[1];
 
-- 
1.7.10.4

