From be3ede4cad2a17a68bf879858781aeb76c2d9b68 Mon Sep 17 00:00:00 2001
From: Andreas Monzner <andreas.monzner@dream-property.net>
Date: Tue, 8 Nov 2016 17:46:25 +0100
Subject: [PATCH] ignore FIGETBSZ ioctl on char devices to allow ioctls with
 cmd 0x02

---
 fs/ioctl.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/fs/ioctl.c b/fs/ioctl.c
index 8ac3fad..cadc7ee 100644
--- a/fs/ioctl.c
+++ b/fs/ioctl.c
@@ -589,7 +589,9 @@ int do_vfs_ioctl(struct file *filp, unsigned int fd, unsigned int cmd,
 		return ioctl_fiemap(filp, arg);
 
 	case FIGETBSZ:
-		return put_user(inode->i_sb->s_blocksize, argp);
+		if (!S_ISCHR(inode->i_mode))
+			return put_user(inode->i_sb->s_blocksize, argp);
+		/* fall through to allow the usage of ioctl's with cmd 0x02 */
 
 	default:
 		if (S_ISREG(inode->i_mode))
-- 
2.1.4

