From 3da9751679b2b954c0e0c480c4b2260e29498ee9 Mon Sep 17 00:00:00 2001
From: Andreas Oberritter <obi@opendreambox.org>
Date: Mon, 18 Aug 2014 01:35:17 +0200
Subject: [PATCH] stat_cache: fix reading block devices

Signed-off-by: Andreas Oberritter <obi@opendreambox.org>
---
 src/stat_cache.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/src/stat_cache.c b/src/stat_cache.c
index 480aae4..45618d3 100644
--- a/src/stat_cache.c
+++ b/src/stat_cache.c
@@ -5,6 +5,7 @@
 
 #include <sys/types.h>
 #include <sys/stat.h>
+#include <sys/mount.h>
 
 #include <stdlib.h>
 #include <string.h>
@@ -617,6 +618,15 @@ handler_t stat_cache_get_entry(server *srv, connection *con, buffer *name, stat_
 		etag_create(sce->etag, &(sce->st), con->etag_flags);
 	} else if (S_ISDIR(st.st_mode)) {
 		etag_create(sce->etag, &(sce->st), con->etag_flags);
+	} else if (S_ISBLK(st.st_mode)) {
+		/* set size for block devices */
+		fd = open(name->ptr, O_RDONLY);
+		if (fd >= 0) {
+			uint64_t size;
+			if (ioctl(fd, BLKGETSIZE64, &size) == 0)
+				sce->st.st_size = size;
+			close(fd);
+		}
 	}
 
 #ifdef HAVE_FAM_H
-- 
1.9.1

