From 04a0c2cb79d92c6c8257bddfb730477567e99d86 Mon Sep 17 00:00:00 2001
From: ghost <andreas.monzner@dream-property.net>
Date: Wed, 11 May 2016 13:49:17 +0200
Subject: [PATCH 3/3] disable window animations for dm520 by default..

---
 usr/share/enigma2/defaults/Dream/settings.520 | 1 +
 1 file changed, 1 insertion(+)

diff --git a/usr/share/enigma2/defaults/Dream/settings.520 b/usr/share/enigma2/defaults/Dream/settings.520
index 96332c2..ec1144e 100644
--- a/usr/share/enigma2/defaults/Dream/settings.520
+++ b/usr/share/enigma2/defaults/Dream/settings.520
@@ -1,2 +1,3 @@
 config.misc.rcused=2
 config.skin.primary_skin=Default-HD/skin.xml
+config.osd.window_animation_default=_-disabled-_
-- 
1.9.1

