diff -Naur gst-plugins-good-1.10.4.orig/ext/soup/gstsouphttpsrc.c gst-plugins-good-1.10.4/ext/soup/gstsouphttpsrc.c
--- gst-plugins-good-1.10.4.orig/ext/soup/gstsouphttpsrc.c	2017-02-22 15:20:59.000000000 +0100
+++ gst-plugins-good-1.10.4/ext/soup/gstsouphttpsrc.c	2022-01-14 23:51:23.890829846 +0100
@@ -937,11 +937,7 @@
     GST_DEBUG_OBJECT (src, "Re-using session");
   }
 
-  if (src->compress)
-    soup_session_add_feature_by_type (src->session, SOUP_TYPE_CONTENT_DECODER);
-  else
-    soup_session_remove_feature_by_type (src->session,
-        SOUP_TYPE_CONTENT_DECODER);
+  soup_session_add_feature_by_type (src->session, SOUP_TYPE_CONTENT_DECODER);
 
   return TRUE;
 }
@@ -1419,6 +1415,11 @@
     }
   }
 
+  if (!src->compress) {
+    soup_message_headers_append (src->msg->request_headers, "Accept-Encoding",
+        "identity");
+  }
+
   soup_message_set_flags (src->msg, SOUP_MESSAGE_OVERWRITE_CHUNKS |
       (src->automatic_redirect ? 0 : SOUP_MESSAGE_NO_REDIRECT));
 
