From 540dd3b5ab2e6276977d8460a6e02f1011f9329d Mon Sep 17 00:00:00 2001
From: Andreas Oberritter <obi@opendreambox.org>
Date: Mon, 23 Apr 2012 01:57:15 +0200
Subject: [PATCH 2/6] fix missing lines on every page on pro7 sat1

---
 src/dvb_demux.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/dvb_demux.c b/src/dvb_demux.c
index f5f6817..b670fc6 100644
--- a/src/dvb_demux.c
+++ b/src/dvb_demux.c
@@ -957,8 +957,8 @@ extract_data_units		(struct frame *		f,
 			if (unlikely (0 != err))
 				goto failed;
 
-			if (unlikely (16 != s->line))
-				goto bad_line;
+/*			if (unlikely (16 != s->line))
+				goto bad_line;*/
 
 			s->id = (0 == f->last_field) ?
 				VBI_SLICED_VPS : VBI_SLICED_VPS_F2;
-- 
1.7.9.5

