From 591f65e8a6612623b9d8320a426b7f2ec2c13f0c Mon Sep 17 00:00:00 2001
From: Andreas Oberritter <obi@opendreambox.org>
Date: Mon, 23 Apr 2012 02:00:29 +0200
Subject: [PATCH 4/6] page catching: add pagenumber

---
 src/search.c |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/src/search.c b/src/search.c
index f9163eb..e240bca 100644
--- a/src/search.c
+++ b/src/search.c
@@ -87,6 +87,10 @@ highlight(struct vbi_search *s, cache_page *vtp,
 	s->row[0] = LAST_ROW + 1;
 	s->col[0] = 0;
 
+	pg->nav_index[0] = first[ms + 0];
+	pg->nav_index[1] = first[ms + 1];
+	pg->nav_index[2] = first[ms + 2];
+
 	for (i = FIRST_ROW; i < LAST_ROW; i++) {
 		vbi_char *acp = &pg->text[i * pg->columns];
 
-- 
1.7.5.4

