From a166d7d80a6e0ae740a36c38b65f49c739b035da Mon Sep 17 00:00:00 2001
From: Andreas Oberritter <obi@opendreambox.org>
Date: Mon, 23 Apr 2012 02:00:51 +0200
Subject: [PATCH 5/7] page catching: search only current page

---
 src/search.c |   12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/src/search.c b/src/search.c
index e240bca..300a74c 100644
--- a/src/search.c
+++ b/src/search.c
@@ -213,7 +213,7 @@ search_page_fwd(cache_page *vtp, vbi_bool wrapped, void *p)
 	flags = 0;
 
 	if (row > LAST_ROW)
-		return 0; /* try next page */
+		return -1; /* try next page */
 
 	for (i = FIRST_ROW; i < LAST_ROW; i++) {
 		acp = &s->pg.text[i * s->pg.columns];
@@ -244,7 +244,7 @@ search_page_fwd(cache_page *vtp, vbi_bool wrapped, void *p)
 	/* Search */
 
 	if (first >= hp)
-		return 0; /* try next page */
+		return -1; /* try next page */
 /*
 fprintf(stderr, "exec: %x/%x; start %d,%d; %c%c%c...\n",
 	vtp->pgno, vtp->subno,
@@ -255,7 +255,7 @@ fprintf(stderr, "exec: %x/%x; start %d,%d; %c%c%c...\n",
 );
 */
 	if (!ure_exec(s->ud, flags, first, hp - first, &ms, &me))
-		return 0; /* try next page */
+		return -1; /* try next page */
 
 	highlight(s, vtp, first, ms, me);
 
@@ -283,7 +283,7 @@ search_page_rev(cache_page *vtp, vbi_bool wrapped, void *p)
 		return -1; /* all done, abort */
 
 	if (vtp->function != PAGE_FUNCTION_LOP)
-		return 0; /* try next page */
+		return -1; /* try next page */
 
 	if (!vbi_format_vt_page(s->vbi, &s->pg, vtp, s->vbi->vt.max_level, 25, 1))
 		return -3; /* formatting error, abort */
@@ -338,7 +338,7 @@ search_page_rev(cache_page *vtp, vbi_bool wrapped, void *p)
 break2:
 
 	if (hp <= s->haystack)
-		return 0; /* try next page */
+		return -1; /* try next page */
 
 	/* Search */
 
@@ -363,7 +363,7 @@ fprintf(stderr, "exec: %x/%x; %d, %d; '%c%c%c...'\n",
 	}
 
 	if (i == 0)
-		return 0; /* try next page */
+		return -1; /* try next page */
 
 	highlight(s, vtp, s->haystack, ms, me);
 
-- 
1.7.9.5

