From d26bd6db22597cdd4050c742bd5569e035e7a2e3 Mon Sep 17 00:00:00 2001
From: Andreas Oberritter <obi@opendreambox.org>
Date: Mon, 23 Apr 2012 01:51:58 +0200
Subject: [PATCH 1/9] Fix linker errors due to implicit declaration of S_ISCHR

Signed-off-by: Andreas Oberritter <obi@opendreambox.org>
---
 contrib/ntsc-cc.c | 1 +
 src/io-dvb.c      | 1 +
 2 files changed, 2 insertions(+)

diff --git a/contrib/ntsc-cc.c b/contrib/ntsc-cc.c
index 066ef07..788e92a 100644
--- a/contrib/ntsc-cc.c
+++ b/contrib/ntsc-cc.c
@@ -33,6 +33,7 @@
 #include <errno.h>
 #include <locale.h>
 #include <sys/ioctl.h>
+#include <sys/stat.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <sys/time.h>
diff --git a/src/io-dvb.c b/src/io-dvb.c
index 2884778..52177b8 100644
--- a/src/io-dvb.c
+++ b/src/io-dvb.c
@@ -28,6 +28,7 @@
 #include <unistd.h>		/* read() */
 #include <errno.h>
 #include <sys/select.h>
+#include <sys/stat.h>
 #include <sys/ioctl.h>
 #include <sys/types.h>
 #include <sys/stat.h>
-- 
1.8.3.2

