From fd041fa0c3544ea6bf82dcce1a32ad5ba07acb59 Mon Sep 17 00:00:00 2001
From: Andreas Oberritter <obi@opendreambox.org>
Date: Mon, 23 Apr 2012 01:51:58 +0200
Subject: [PATCH 1/6] Fix linker errors due to implicit declaration of S_ISCHR

Signed-off-by: Andreas Oberritter <obi@opendreambox.org>
---
 contrib/ntsc-cc.c |    1 +
 src/io-dvb.c      |    1 +
 2 files changed, 2 insertions(+)

diff --git a/contrib/ntsc-cc.c b/contrib/ntsc-cc.c
index a89a5f9..3bf6680 100644
--- a/contrib/ntsc-cc.c
+++ b/contrib/ntsc-cc.c
@@ -33,6 +33,7 @@
 #include <errno.h>
 #include <locale.h>
 #include <sys/ioctl.h>
+#include <sys/stat.h>
 #include <sys/types.h>
 #include <sys/time.h>
 #ifdef HAVE_GETOPT_LONG
diff --git a/src/io-dvb.c b/src/io-dvb.c
index bc5d634..4405b87 100644
--- a/src/io-dvb.c
+++ b/src/io-dvb.c
@@ -28,6 +28,7 @@
 #include <unistd.h>		/* read() */
 #include <errno.h>
 #include <sys/select.h>
+#include <sys/stat.h>
 #include <sys/ioctl.h>
 
 #ifndef HAVE_S64_U64
-- 
1.7.9.5

