From f9a4433f9f5951cd446e24145639b1e1c208d79a Mon Sep 17 00:00:00 2001
From: Andreas Oberritter <obi@opendreambox.org>
Date: Mon, 23 Apr 2012 02:01:29 +0200
Subject: [PATCH 4/9] fix sky not working

---
 src/dvb_demux.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/dvb_demux.c b/src/dvb_demux.c
index 1855518..0ce10ea 100644
--- a/src/dvb_demux.c
+++ b/src/dvb_demux.c
@@ -1448,8 +1448,8 @@ valid_vbi_pes_packet_header	(vbi_dvb_demux *	dx,
 			"PTS missing in PES header.");
 
 		/* XXX make this optional to handle broken sources. */
-		if (dx->new_frame)
-			return FALSE;
+/*		if (dx->new_frame)
+			return FALSE;*/
 
 		break;
 	}
-- 
1.8.3.2

