diff -Naur enigma2_4.2.0r1_dm7080.orig/lib/systemd/system/enigma2.service enigma2_4.2.0r1_dm7080/lib/systemd/system/enigma2.service
--- enigma2_4.2.0r1_dm7080.orig/lib/systemd/system/enigma2.service	2014-09-05 01:09:28.000000000 +0200
+++ enigma2_4.2.0r1_dm7080/lib/systemd/system/enigma2.service	2014-09-05 20:37:31.297228884 +0200
@@ -12,10 +12,9 @@
 ExecStopPost=/usr/bin/enigma2-system-state
 NotifyAccess=main
 OOMScoreAdjust=-999
-Restart=always
-RestartPreventExitStatus=2 8
+Restart=on-failure
 RestartSec=0
-SuccessExitStatus=2 3 8
+SuccessExitStatus=2 8
 Type=notify
 
 [Install]
diff -Naur enigma2_4.2.0r1_dm7080.orig/usr/bin/enigma2-system-state enigma2_4.2.0r1_dm7080/usr/bin/enigma2-system-state
--- enigma2_4.2.0r1_dm7080.orig/usr/bin/enigma2-system-state	2014-09-05 01:09:28.000000000 +0200
+++ enigma2_4.2.0r1_dm7080/usr/bin/enigma2-system-state	2014-09-05 23:39:06.382982710 +0200
@@ -3,10 +3,10 @@
 	state=`cat /run/enigma2/system-state`
 	case "$state" in
 		"system-restart")
-			/sbin/reboot
+			systemctl --no-block reboot
 			;;
 		"system-standby")
-			/sbin/poweroff
+			systemctl --no-block poweroff
 			;;
 		"ui-restart")
 			;;
