From 1d3b4e7b5ce41003fadf52ef0ad69ed2cd4bdc6c Mon Sep 17 00:00:00 2001
From: ghost <andreas.monzner@dream-property.net>
Date: Mon, 9 Feb 2015 15:11:36 +0100
Subject: [PATCH 1/2] config.py: fixed possible crash

---
 lib/python/Components/config.py |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/usr/lib/enigma2/python/Components/config.py b/usr/lib/enigma2/python/Components/config.py
index 5ac6d14..ae9978e 100644
--- a/usr/lib/enigma2/python/Components/config.py
+++ b/usr/lib/enigma2/python/Components/config.py
@@ -969,7 +969,7 @@ class ConfigTextBase(ConfigElement):
 
 	def genText(self):
 		tlen = len(self.text)
-		if self.censor_char != '':
+		if tlen and self.censor_char != '':
 			tmp = [ self.censor_char for c in range(tlen) ]
 			if self.selected:
 				mpos = self.marked_pos
-- 
1.7.10.4

