--- a/liveMedia/RTSPServer.cpp	2018-02-18 09:53:11.000000000 +0100
+++ b/liveMedia/RTSPServer.cpp	2018-03-28 13:32:23.000000000 +0200
@@ -1803,16 +1803,7 @@
 void RTSPServer::RTSPClientSession
 ::handleCmd_PAUSE(RTSPServer::RTSPClientConnection* ourClientConnection,
 		  ServerMediaSubsession* subsession) {
-  for (unsigned i = 0; i < fNumStreamStates; ++i) {
-    if (subsession == NULL /* means: aggregated operation */
-	|| subsession == fStreamStates[i].subsession) {
-      if (fStreamStates[i].subsession != NULL) {
-	fStreamStates[i].subsession->pauseStream(fOurSessionId, fStreamStates[i].streamToken);
-      }
-    }
-  }
-  
-  setRTSPResponse(ourClientConnection, "200 OK", fOurSessionId);
+  handleCmd_TEARDOWN(ourClientConnection, subsession);
 }
 
 void RTSPServer::RTSPClientSession
