From 97f3b955b9af2c4023ff014d42c2d5b87e6c4795 Mon Sep 17 00:00:00 2001
From: Denys Vlasenko <vda.linux@googlemail.com>
Date: Fri, 22 Jun 2012 15:06:27 +0200
Subject: [PATCH 6/8] mount: undo recent breakage when mount flags were made
 unsigned

Signed-off-by: Denys Vlasenko <vda.linux@googlemail.com>
---
 util-linux/mount.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/util-linux/mount.c b/util-linux/mount.c
index 220a4e6..95dee18 100644
--- a/util-linux/mount.c
+++ b/util-linux/mount.c
@@ -467,7 +467,7 @@ static unsigned long parse_mount_options(char *options, char **unrecognized)
 		for (i = 0; i < ARRAY_SIZE(mount_options); i++) {
 			if (strcasecmp(option_str, options) == 0) {
 				unsigned long fl = mount_options[i];
-				if (fl < 0)
+				if ((long)fl < 0)
 					flags &= fl;
 				else
 					flags |= fl;
-- 
1.7.10.4

