From a367c1230a5320c4195b99c681e0c5575c7dfb81 Mon Sep 17 00:00:00 2001
From: Andreas Oberritter <obi@opendreambox.org>
Date: Sun, 3 Aug 2014 20:30:22 +0200
Subject: [PATCH 8/8] speed_table: add baud rates up to 4 MBit/s as defined by
 glibc

Signed-off-by: Andreas Oberritter <obi@opendreambox.org>
---
 libbb/speed_table.c | 30 ++++++++++++++++++++++++++++++
 1 file changed, 30 insertions(+)

diff --git a/libbb/speed_table.c b/libbb/speed_table.c
index 941e517..241e618 100644
--- a/libbb/speed_table.c
+++ b/libbb/speed_table.c
@@ -51,9 +51,39 @@ static const struct speed_map speeds[] = {
 #ifdef B460800
 	{B460800, 460800},
 #endif
+#ifdef B500000
+	{B500000, 500000},
+#endif
+#ifdef B576000
+	{B576000, 576000},
+#endif
 #ifdef B921600
 	{B921600, 921600},
 #endif
+#ifdef B1000000
+	{B1000000, 1000000},
+#endif
+#ifdef B1152000
+	{B1152000, 1152000},
+#endif
+#ifdef B1500000
+	{B1500000, 1500000},
+#endif
+#ifdef B2000000
+	{B2000000, 2000000},
+#endif
+#ifdef B2500000
+	{B2500000, 2500000},
+#endif
+#ifdef B3000000
+	{B3000000, 3000000},
+#endif
+#ifdef B3500000
+	{B3500000, 3500000},
+#endif
+#ifdef B4000000
+	{B4000000, 4000000},
+#endif
 };
 
 enum { NUM_SPEEDS = ARRAY_SIZE(speeds) };
-- 
1.9.1

