From ec5f32a309666dd6ad75f7a3cb1faf148d0366e7 Mon Sep 17 00:00:00 2001
From: Andreas Oberritter <obi@opendreambox.org>
Date: Thu, 11 Apr 2019 14:46:44 +0200
Subject: [PATCH] webadmin: remove remnants of tpm check

---
 webadmin/src/__init__.py | 27 ---------------------------
 1 file changed, 27 deletions(-)

diff --git a/webadmin/src/__init__.py b/webadmin/src/__init__.py
index 389af701..3b4d1c14 100644
--- a/webadmin/src/__init__.py
+++ b/webadmin/src/__init__.py
@@ -1,28 +1 @@
-import sha
-
 __version__ = "1.6.6"
-
-def bin2long(s):
-	return reduce( lambda x,y:(x<<8L)+y, map(ord, s))
-
-def long2bin(l):
-	res = ""
-	for byte in range(128):
-		res += chr((l >> (1024 - (byte + 1) * 8)) & 0xff)
-	return res
-
-def rsa_pub1024(src, mod):
-	return long2bin(pow(bin2long(src), 65537, bin2long(mod)))
-	
-def decrypt_block(src, mod):
-	if len(src) != 128 and len(src) != 202:
-		return None
-	dest = rsa_pub1024(src[:128], mod)
-	hash = sha.new(dest[1:107])
-	if len(src) == 202:
-		hash.update(src[131:192])	
-	result = hash.digest()
-	if result == dest[107:127]:
-		return dest
-	return None
-
