From 9bf30869244e82fbecb8c447b81de47312ec569a Mon Sep 17 00:00:00 2001
From: Andreas Monzner <andreas.monzner@dream-property.net>
Date: Thu, 10 Dec 2020 18:01:08 +0100
Subject: [PATCH 8/8] Autoresolution: fixed handling for enigma2 >= 4.5.1r3

---
 autoresolution/src/plugin.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/autoresolution/src/plugin.py b/autoresolution/src/plugin.py
index 5485459..4d1296a 100644
--- a/autoresolution/src/plugin.py
+++ b/autoresolution/src/plugin.py
@@ -245,8 +245,7 @@ class AutoRes(Screen):
 						self.lastmode = new_mode
 						self.changeVideomode()
 					elif old != resolutionlabel["content"].getText() and config.plugins.autoresolution.showinfo.value:
-						# sometimes mode is changed via displaymanager.. so we always call setMode here to update the label
-						self.setMode(new_mode, False)
+						self.setMode(new_mode)
 						resolutionlabel.show()
 
 	def changeVideomode(self):
@@ -275,6 +274,8 @@ class AutoRes(Screen):
 	def setMode(self, mode, set=True):
 		port_txt = "HDMI" if port == "DVI" else port
 
+		self.lastmode = mode
+
 		if mode.find("0p30") != -1 or mode.find("0p24") != -1 or mode.find("0p25") != -1:
 			match = re.search(r"(\d*?[ip])(\d*?)$", mode)
 			mode = match.group(1)
@@ -288,7 +289,6 @@ class AutoRes(Screen):
 			if config.plugins.autoresolution.showinfo.value:
 				resolutionlabel.show()
 			DisplayHardware.instance.setMode(port, mode, rate)
-		self.lastmode = mode
 
 class ResolutionLabel(Screen):
 	height = getDesktop(0).size().height()
-- 
2.1.4

