diff -Naur a/lib/systemd/system/enigma2-environment.service b/lib/systemd/system/enigma2-environment.service
--- a/lib/systemd/system/enigma2-environment.service	2019-08-22 19:05:00.000000000 +0200
+++ b/lib/systemd/system/enigma2-environment.service	2019-08-23 12:19:11.067734323 +0200
@@ -2,6 +2,8 @@
 Description=Generate environment for enigma2
 Before=enigma2.service
 DefaultDependencies=no
+Requires=dev-dvb-adapter0-video0.device
+After=dev-dvb-adapter0-video0.device
 
 [Service]
 Type=oneshot
