diff -Naur cdfs-2.6.27.orig/root.c cdfs-2.6.27/root.c
--- cdfs-2.6.27.orig/root.c	2016-07-13 17:09:54.000000000 +0200
+++ cdfs-2.6.27/root.c	2016-07-13 17:16:19.936431303 +0200
@@ -545,9 +545,13 @@
   PRINT("this_cd = 0x%x\n", (unsigned)this_cd);
 
   PRINT("read inode %ld\n", i->i_ino);
-  
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3,14,0)
+  i->i_uid.val    = this_cd->uid;
+  i->i_gid.val    = this_cd->gid;
+#else
   i->i_uid        = this_cd->uid;
   i->i_gid        = this_cd->gid;
+#endif
   set_nlink(i, 1);
   i->i_op         = &cdfs_inode_operations;
   i->i_fop        = NULL;
