--- git/ext/faad/gstfaad.c	2014-03-20 14:21:21.000000000 +0100
+++ git/ext/faad/gstfaad.c	2014-03-20 15:14:44.000000000 +0100
@@ -867,7 +867,7 @@
 static gboolean
 plugin_init (GstPlugin * plugin)
 {
-  return gst_element_register (plugin, "faad", GST_RANK_PRIMARY, GST_TYPE_FAAD);
+  return gst_element_register (plugin, "faad", GST_RANK_SECONDARY, GST_TYPE_FAAD);
 }
 
 GST_PLUGIN_DEFINE (GST_VERSION_MAJOR,
