--- a/usr/lib/enigma2/python/Tools/Downloader.py
+++ b/usr/lib/enigma2/python/Tools/Downloader.py
@@ -2,8 +2,8 @@ from twisted.web import client
 from twisted.internet import reactor, defer
 
 class HTTPProgressDownloader(client.HTTPDownloader):
-	def __init__(self, url, outfile, headers=None):
-		client.HTTPDownloader.__init__(self, url, outfile, headers=headers, agent="Dreambox HTTP Downloader")
+	def __init__(self, url, outfile, headers=None, agent="Dreambox HTTP Downloader"):
+		client.HTTPDownloader.__init__(self, url, outfile, headers=headers, agent=agent)
 		self.status = None
 		self.progress_callback = None
 		self.deferred = defer.Deferred()
