diff -Naur enigma2_3.999git20140624_mips32el.orig/usr/bin/enigma2.sh enigma2_3.999git20140624_mips32el/usr/bin/enigma2.sh
--- enigma2_3.999git20140624_mips32el.orig/usr/bin/enigma2.sh	2014-06-24 19:23:34.000000000 +0200
+++ enigma2_3.999git20140624_mips32el/usr/bin/enigma2.sh	2014-09-15 16:33:53.454025786 +0200
@@ -13,8 +13,8 @@
 	[ -x $i ] && $i || /bin/true
 done
 
-if [ -d /home/root ]; then
-	cd /home/root
+if [ -d /root ]; then
+	cd /root
 fi
 
 LIBS="$LIBS /usr/lib/libopen.so"
@@ -48,25 +48,25 @@
 ret=$?
 case $ret in
 	1)
-		/sbin/halt
+		systemctl --no-block poweroff
 		;;
 	2)
-		/sbin/reboot
+		systemctl --no-block reboot
 		;;
 	4)
 		/sbin/rmmod lcd
-		/usr/sbin/fpupgrade --upgrade 2>&1 | tee /home/root/fpupgrade.log
+		/usr/sbin/fpupgrade --upgrade 2>&1 | tee /root/fpupgrade.log
 		sleep 1;
 		/sbin/rmmod fp
 		/sbin/modprobe fp
-		/sbin/reboot
+		systemctl --no-block reboot
 		;;
 	6)
 		/usr/bin/opkgfb
 		;;
 	7)
 		/usr/bin/opkgfb
-		/sbin/reboot
+		systemctl --no-block reboot
 		;;
 	*)
 		;;
