From d9c7bdd4ef4012d85b7a766aff72cb133c041c12 Mon Sep 17 00:00:00 2001
From: Andreas Oberritter <obi@opendreambox.org>
Date: Sat, 13 Sep 2014 16:42:51 +0200
Subject: [PATCH 1/4] build: make man2html optional

Signed-off-by: Andreas Oberritter <obi@opendreambox.org>
---
 configure.ac    | 4 +---
 doc/Makefile.am | 5 ++++-
 2 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/configure.ac b/configure.ac
index 8a58c69..57fa900 100644
--- a/configure.ac
+++ b/configure.ac
@@ -20,9 +20,7 @@ AC_PATH_PROG(mknod, mknod, /bin/mknod)
 AC_PATH_PROG(mkfifo, mkfifo, /usr/bin/mkfifo)
 AC_PATH_PROG(depmod, depmod, /sbin/depmod, $PATH:/sbin)
 AC_CHECK_PROG([MAN2HTML],[man2html],[yes],[no])
-if test x$MAN2HTML = xno; then
-    AC_MSG_ERROR(Cannot find required program man2html)
-fi
+AM_CONDITIONAL(HAVE_MAN2HTML, test x$MAN2HTML = xyes)
 AC_PROG_LN_S
 LT_INIT([disable-static])
 AM_PATH_PYTHON([3.1],,[:])
diff --git a/doc/Makefile.am b/doc/Makefile.am
index 8dcf728..57387ab 100644
--- a/doc/Makefile.am
+++ b/doc/Makefile.am
@@ -17,7 +17,10 @@ docdir                  = @datadir@/doc/@PACKAGE@
 dist_doc_DATA           = lirc.css irxevent.keys
 
 htmldir                 = $(docdir)/html
-nodist_html_DATA        = $(generated_html) $(manpage_html)
+nodist_html_DATA        = $(generated_html)
+if HAVE_MAN2HTML
+nodist_html_DATA        += $(manpage_html)
+endif
 dist_html_DATA          = $(html_includes) html-source/lirc.css table.html
 
 dist_noinst_DATA        = $(man_sources) $(html_sources)
-- 
1.9.1

