From be0858371e3d90501370c66169ec5a1b7fc9f810 Mon Sep 17 00:00:00 2001
From: Andreas Oberritter <obi@opendreambox.org>
Date: Sun, 14 Sep 2014 20:25:41 +0200
Subject: [PATCH 4/4] build: only link to systemd library where required

Signed-off-by: Andreas Oberritter <obi@opendreambox.org>
---
 configure.ac        | 2 --
 daemons/Makefile.am | 7 ++++---
 2 files changed, 4 insertions(+), 5 deletions(-)

diff --git a/configure.ac b/configure.ac
index 01fb3be..7d50c61 100644
--- a/configure.ac
+++ b/configure.ac
@@ -295,8 +295,6 @@ AC_CHECK_HEADERS(linux/i2c-dev.h,[
 
 
 PKG_CHECK_MODULES([SYSTEMD],[libsystemd-daemon],[AC_DEFINE(HAVE_SYSTEMD)],[true])
-CFLAGS="$CFLAGS $SYSTEMD_CFLAGS"
-LIBS="$LIBS $SYSTEMD_LIBS"
 
 SYSTEMD_SYSTEMUNITDIR
 
diff --git a/daemons/Makefile.am b/daemons/Makefile.am
index 69f9fe2..15199a1 100644
--- a/daemons/Makefile.am
+++ b/daemons/Makefile.am
@@ -2,6 +2,7 @@
 ## Process this file with automake to produce Makefile.in
 ACLOCAL_AMFLAGS         = -I m4
 
+AM_CFLAGS               = $(SYSTEMD_CFLAGS)
 AM_CPPFLAGS             = -I$(top_srcdir) -I$(top_srcdir)/lib -Wall
 AM_LDFLAGS              = -lpthread
 
@@ -12,7 +13,7 @@ varrundir               = $(localstatedir)/run
 sbin_PROGRAMS           = lircd lircmd
 
 lircd_SOURCES           = lircd.c
-lircd_LDADD             = ../lib/liblirc.la
+lircd_LDADD             = ../lib/liblirc.la $(SYSTEMD_LIBS)
 
 lircmd_SOURCES          = lircmd.c
 lircmd_LDADD            = ../lib/liblirc.la
@@ -20,10 +21,10 @@ lircmd_LDADD            = ../lib/liblirc.la
 noinst_PROGRAMS         = lircd.simsend lircd.simrec
 lircd_simsend_SOURCES   = lircd.c
 lircd_simsend_CFLAGS    = -DSIM_SEND
-lircd_simsend_LDADD     = ../lib/liblirc.la
+lircd_simsend_LDADD     = ../lib/liblirc.la $(SYSTEMD_LIBS)
 lircd_simrec_SOURCES    = lircd.c
 lircd_simrec_CFLAGS     = -DSIM_REC
-lircd_simrec_LDADD      = ../lib/liblirc.la
+lircd_simrec_LDADD      = ../lib/liblirc.la $(SYSTEMD_LIBS)
 
 
 if USE_FIFOS
-- 
1.9.1

