From f27eda0d1690d3a86bc3d3fcc4289eefc5af0ba6 Mon Sep 17 00:00:00 2001
From: ghost <andreas.monzner@dream-property.net>
Date: Wed, 21 Dec 2016 22:54:39 +0100
Subject: [PATCH] add support for dm520/dm900 reboot workaround

---
 usr/bin/enigma2-system-state | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/usr/bin/enigma2-system-state b/usr/bin/enigma2-system-state
index eb3e2a7..bbca05b 100644
--- a/usr/bin/enigma2-system-state
+++ b/usr/bin/enigma2-system-state
@@ -3,7 +3,12 @@ if [ -f /run/enigma2/system-state ]; then
 	state=`cat /run/enigma2/system-state`
 	case "$state" in
 		"system-restart")
-			systemctl --no-block reboot
+			if [ -f /proc/stb/fp/force_restart ]; then
+				echo 1 > /proc/stb/fp/force_restart
+				systemctl --no-block poweroff
+			else
+				systemctl --no-block reboot
+			fi
 			;;
 		"system-standby")
 			systemctl --no-block poweroff
-- 
2.1.4

