From a6b52ef009458f8b53a705a8936ed63b8952bb5d Mon Sep 17 00:00:00 2001
From: ghost <andreas.monzner@dream-property.net>
Date: Tue, 1 Sep 2015 17:04:54 +0200
Subject: [PATCH] add hack to disable PGS / VOB subtitles

---
 usr/lib/enigma2/python/mytest.py | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/usr/lib/enigma2/python/mytest.py b/usr/lib/enigma2/python/mytest.py
index 12197ae..f02024f 100644
--- a/usr/lib/enigma2/python/mytest.py
+++ b/usr/lib/enigma2/python/mytest.py
@@ -60,6 +60,20 @@ def ePositionGauge_setPointer(self, which, pixmap, center):
 		ePositionGauge_setPointer_org(self, which, pixmap, center)
 enigma.ePositionGauge.setPointer = ePositionGauge_setPointer
 
+from enigma import stPGS, stVOB
+
+iSubtitleOutputPtr_getSubtitleList_org = enigma.iSubtitleOutputPtr.getSubtitleList
+def iSubtitleOutputPtr_getSubtitleList(self):
+	lst_filtered = [ ]
+	for x in iSubtitleOutputPtr_getSubtitleList_org(self):
+		if x[0] == 2 and x[2] in (stPGS, stVOB):
+			print "PGS/VOB subtitles broken... ignore", x
+		else:
+			lst_filtered.append(x)
+	return lst_filtered
+
+enigma.iSubtitleOutputPtr.getSubtitleList = iSubtitleOutputPtr_getSubtitleList
+
 from Tools.Profile import profile, profile_final
 
 profile("PYTHON_START")
-- 
1.9.1

