From 0319cce0ac1340e5ba4427249ae4f2f12025851c Mon Sep 17 00:00:00 2001
From: Stephan Reichholf <reichi@opendreambox.org>
Date: Sat, 4 Mar 2017 12:55:01 +0100
Subject: [PATCH 3/3] vtuner: send tune status messages

---
 vtuner.cpp | 50 ++++++++++++++++++++++++++++++++++++++++++++++++--
 vtuner.h   |  4 ++++
 2 files changed, 52 insertions(+), 2 deletions(-)

diff --git a/vtuner.cpp b/vtuner.cpp
index b1fcb52..c93c934 100644
--- a/vtuner.cpp
+++ b/vtuner.cpp
@@ -25,6 +25,7 @@
 #include <sys/ioctl.h>
 #include <sys/types.h>
 #include <sys/stat.h>
+#include <time.h>
 #include <unistd.h>
 #include <fcntl.h>
 
@@ -36,6 +37,42 @@
 
 const char* vtuner_path = "/dev/misc/vtuner";
 
+static inline timespec operator-( const timespec &t1, const timespec &t2 )
+{
+	timespec tmp;
+	tmp.tv_sec = t1.tv_sec - t2.tv_sec;
+	if ( (tmp.tv_nsec = t1.tv_nsec - t2.tv_nsec) < 0 )
+	{
+		tmp.tv_sec--;
+		tmp.tv_nsec += 1000000000;
+	}
+	return tmp;
+}
+
+static inline timespec &operator+=( timespec &t1, const long msek )
+{
+	t1.tv_sec += msek / 1000;
+	if ( (t1.tv_nsec += (msek % 1000) * 1000000) >= 1000000000 )
+	{
+		t1.tv_sec++;
+		t1.tv_nsec -= 1000000000;
+	}
+	return t1;
+}
+
+static inline int64_t timeout_msec ( const timespec & orig, const timespec &now )
+{
+	const timespec tv = orig - now;
+	return (int64_t)tv.tv_sec * 1000 + tv.tv_nsec / 1000000;
+}
+
+static inline int64_t timeout_msec ( const timespec & orig )
+{
+	timespec now;
+	clock_gettime(CLOCK_MONOTONIC, &now);
+	return timeout_msec(orig, now);
+}
+
 satipVtuner::satipVtuner(satipConfig* satip_cfg)
 	:m_fd(-1), m_openok(false), m_tone(SEC_TONE_OFF)
 {
@@ -473,6 +510,7 @@ void satipVtuner::vtunerEvent()
 #else
 			DEBUG(MSG_MAIN,"MSG_SET_FRONTEND skip..\n");
 #endif
+			++tune_tries;
 			break;
 
 		case MSG_GET_FRONTEND:
@@ -496,11 +534,16 @@ void satipVtuner::vtunerEvent()
 			break;
 
 		case MSG_READ_STATUS:
+			msg.body.status = 0;
 			//INFO(MSG_MAIN,"MSG_READ_STATUS\n");
 			if (m_satip_rtp && m_satip_rtp->getHasLock())
 				msg.body.status = FE_HAS_LOCK;
-			else
-				msg.body.status = 0;
+			else {
+				if (timeout_msec(timeout) < 0)
+					msg.body.status |= FE_TIMEDOUT;
+				if (tune_tries & 4)
+					msg.body.status |= FE_HAS_SIGNAL;
+			}
 			break;
 		case MSG_READ_BER:
 			//INFO(MSG_MAIN,"MSG_READ_BER\n");
@@ -560,6 +603,9 @@ void satipVtuner::vtunerEvent()
 			if (m_satip_rtp) {
 				m_satip_rtp->unset();
 			}
+			tune_tries = 0;
+			clock_gettime(CLOCK_MONOTONIC, &timeout);
+			timeout += 1500;
 			msg.body.tune_settings.min_delay_ms = 50; // let linux-tv api ask every 50ms for locked/unlocked state...
 			msg.body.tune_settings.step_size = 0;
 			msg.body.tune_settings.max_drift = 0;
diff --git a/vtuner.h b/vtuner.h
index 990dec4..8f027e1 100644
--- a/vtuner.h
+++ b/vtuner.h
@@ -169,6 +169,10 @@ class satipVtuner
 {
 	int m_fd;
 	bool m_openok;
+
+	int tune_tries;
+	timespec timeout;
+
 #if VMSG_TYPE1
 	fe_sec_tone_mode_t m_tone;
 #else
-- 
2.7.4

