From 801ae0bc79ee96d6fd535afc9b043f419900e14e Mon Sep 17 00:00:00 2001
From: Andreas Oberritter <obi@opendreambox.org>
Date: Wed, 8 Mar 2017 18:34:34 +0100
Subject: [PATCH] build: fix presets for mips

---
 configure.ac | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/configure.ac b/configure.ac
index 1a60bde..1b4dd46 100644
--- a/configure.ac
+++ b/configure.ac
@@ -652,9 +652,6 @@ case $host in
      CORE_SYSTEM_NAME=linux
      ARCH="mips"
      use_arch="mips"
-     use_gles=no
-     use_gl=yes
-     use_sdl=yes
      USE_STATIC_FFMPEG=1
      ;;
   *)
