From b7f20315000516e023e22ee04a7a952c9ecb7828 Mon Sep 17 00:00:00 2001
From: ghost <andreas.monzner@dream-property.net>
Date: Mon, 1 Feb 2016 13:55:38 +0100
Subject: [PATCH 2/3] fixed mistake

---
 Navigation.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/usr/lib/enigma2/python/Navigation.py b/usr/lib/enigma2/python/Navigation.py
index 02f868c..1c7efa6 100644
--- a/usr/lib/enigma2/python/Navigation.py
+++ b/usr/lib/enigma2/python/Navigation.py
@@ -123,9 +123,6 @@ class Navigation:
 		return service
 
 	def stopRecordService(self, service):
-		if not self.immediatePlay:
-			self.scheduledServiceReference = None
-			return
 		ret = self.pnav and self.pnav.stopRecordService(service)
 		return ret
 
@@ -138,6 +135,9 @@ class Navigation:
 		return self.currentlyPlayingService
 
 	def stopService(self):
+		if not self.immediatePlay:
+			self.scheduledServiceReference = None
+			return
 		print "stopService"
 		if self.pnav:
 			self.pnav.stopService()
-- 
1.9.1

