From e2f8d10c6c46acc1c2e7d24f01b5d2929505c907 Mon Sep 17 00:00:00 2001
From: Andreas Oberritter <obi@opendreambox.org>
Date: Fri, 10 Oct 2014 22:26:47 +0200
Subject: [PATCH 2/7] mpegdemux: compile fixes

---
 gst/mpegdemux/gstmpegdemux.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/gst/mpegdemux/gstmpegdemux.c b/gst/mpegdemux/gstmpegdemux.c
index 6fe21e7..840ca1c 100644
--- a/gst/mpegdemux/gstmpegdemux.c
+++ b/gst/mpegdemux/gstmpegdemux.c
@@ -292,7 +292,7 @@ gst_flups_demux_reset (GstFluPSDemux * demux)
       if (stream->pad)
         gst_element_remove_pad (GST_ELEMENT_CAST (demux), stream->pad);
       if (stream->buf)
-        free(stream->buf);
+        g_free(stream->buf);
       if (stream->meta_buf)
         gst_buffer_unref(stream->meta_buf);
       g_free (stream);
@@ -429,7 +429,7 @@ gst_flups_demux_create_stream (GstFluPSDemux * demux, gint id, gint stream_type)
   stream->segment_thresh = threshold;
 
   if (threshold == VIDEO_SEGMENT_THRESHOLD) {
-    stream->buf = malloc(64*1024);
+    stream->buf = g_malloc(64*1024);
     stream->buf_pos = 0;
     stream->meta_buf = gst_buffer_new();
   }
@@ -484,7 +484,7 @@ static GstFlowReturn
 gst_flups_demux_send_data (GstFluPSDemux * demux, GstFluPSStream * stream,
     GstBuffer * buf)
 {
-  GstFlowReturn result;
+  GstFlowReturn result = GST_FLOW_OK;
   guint64 timestamp;
   guint size;
   gboolean sent = FALSE;
-- 
1.9.1

