diff -Naur org/ext/rtmp/gstrtmpsrc.c git/ext/rtmp/gstrtmpsrc.c
--- org/ext/rtmp/gstrtmpsrc.c	2013-07-16 17:17:38.000000000 +0200
+++ git/ext/rtmp/gstrtmpsrc.c	2013-07-16 17:20:56.154789765 +0200
@@ -57,8 +57,6 @@
 #include <winsock2.h>
 #endif
 
-#include <sys/socket.h>
-
 GST_DEBUG_CATEGORY_STATIC (rtmpsrc_debug);
 #define GST_CAT_DEFAULT rtmpsrc_debug
 
@@ -84,7 +82,6 @@
     GValue * value, GParamSpec * pspec);
 static void gst_rtmp_src_finalize (GObject * object);
 
-static gboolean gst_rtmp_src_unlock (GstBaseSrc * src);
 static gboolean gst_rtmp_src_stop (GstBaseSrc * src);
 static gboolean gst_rtmp_src_start (GstBaseSrc * src);
 static gboolean gst_rtmp_src_is_seekable (GstBaseSrc * src);
@@ -148,7 +145,6 @@
 
   gstbasesrc_class->start = GST_DEBUG_FUNCPTR (gst_rtmp_src_start);
   gstbasesrc_class->stop = GST_DEBUG_FUNCPTR (gst_rtmp_src_stop);
-  gstbasesrc_class->unlock = GST_DEBUG_FUNCPTR (gst_rtmp_src_unlock);
   gstbasesrc_class->is_seekable = GST_DEBUG_FUNCPTR (gst_rtmp_src_is_seekable);
   gstbasesrc_class->prepare_seek_segment =
       GST_DEBUG_FUNCPTR (gst_rtmp_src_prepare_seek_segment);
@@ -584,24 +580,6 @@
 #undef STR2AVAL
 
 static gboolean
-gst_rtmp_src_unlock (GstBaseSrc * basesrc)
-{
-  GstRTMPSrc *rtmpsrc = GST_RTMP_SRC (basesrc);
-
-  GST_DEBUG_OBJECT (rtmpsrc, "unlock");
-
-  /* This cancels the recv() underlying RTMP_Read, but will cause a
-   * SIGPIPE.  Hopefully the app is ignoring it, or you've patched
-   * librtmp. */
-  if (rtmpsrc->rtmp && rtmpsrc->rtmp->m_sb.sb_socket > 0) {
-    shutdown (rtmpsrc->rtmp->m_sb.sb_socket, SHUT_RDWR);
-  }
-
-  return TRUE;
-}
-
-
-static gboolean
 gst_rtmp_src_stop (GstBaseSrc * basesrc)
 {
   GstRTMPSrc *src;
